test/e2e/node/pods.go

```diff
@@ -415,6 +415,166 @@ var _ = SIGDescribe("Pods Extended", func() {
 		})
 	})
 
+	ginkgo.Describe("Pod Generation", func() {
+		var podClient *e2epod.PodClient
+		ginkgo.BeforeEach(func() {
+			podClient = e2epod.NewPodClient(f)
+		})
+
+		ginkgo.It("pod generation should start at 1 and increment per update", func(ctx context.Context) {
+			ginkgo.By("creating the pod")
+			podName := "pod-generation-" + string(uuid.NewUUID())
+			value := strconv.Itoa(time.Now().Nanosecond())
+			pod := e2epod.NewAgnhostPod(f.Namespace.Name, podName, nil, nil, nil)
+			pod.ObjectMeta.Labels = map[string]string{
+				"time": value,
+			}
+			pod.Spec.InitContainers = []v1.Container{{
+				Name:  "init-container",
+				Image: imageutils.GetE2EImage(imageutils.BusyBox),
+			}}
+
+			ginkgo.By("submitting the pod to kubernetes")
+			pod = podClient.CreateSync(ctx, pod)
+			ginkgo.DeferCleanup(func(ctx context.Context) error {
+				ginkgo.By("deleting the pod")
+				return podClient.Delete(ctx, pod.Name, metav1.DeleteOptions{})
+			})
+
+			ginkgo.By("setting up selector")
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+
+			ginkgo.By("verifying pod generation bumps as expected")
+			tests := []struct {
+				name                 string
+				updateFn             func(*v1.Pod)
+				expectGenerationBump bool
+			}{
+				{
+					name:                 "verifying the new pod's generation is 1",
```

Comment:
nit: call this something like "empty update", and verify the initial generation separately (just assert it's 1 right after creating it)